lib/fast-path.ts

```diff
@@ -384,48 +374,31 @@ FPp.needsParens = function(assumeExpressionContext) {
       return false;
 
     case "ExpressionStatement":
-      return name !== "expression";
+      return parent.expression !== node;
 
     default:
       // Otherwise err on the side of overparenthesization, adding
       // explicit exceptions above if this proves overzealous.
       return true;
     }
 
-  case "YieldExpression":
-    switch (parent.type) {
-    case "BinaryExpression":
-    case "LogicalExpression":
-    case "UnaryExpression":
-    case "SpreadElement":
-    case "SpreadProperty":
-    case "CallExpression":
-    case "MemberExpression":
-    case "NewExpression":
-    case "ConditionalExpression":
```

Comment:
These only require parens in certain positions. Once this and `YieldExpression` are accounted for it is clear that `yield` and `await` behave the same was as assignment and conditional expressions.